classes/Hook/HookActionCarrierProcess.php

```diff
@@ -78,10 +78,12 @@ protected function getUniversalAnalytics(string $carrierName)
 
     protected function getGoogleAnalytics4(string $carrierName)
     {
-        return 'gtag("event", "add_shipping_info", {
-            currency: "' . $this->context->currency->iso_code . '",
-            value: ' . $this->context->cart->getCartTotalPrice() . ',
-            shipping_tier: "' . $carrierName . '"
-          });';
+        $eventData = [
+            'currency' => $this->context->currency->iso_code,
+            'value' => (float) $this->context->cart->getCartTotalPrice(),
```

Comment:
The `getCartTotalPrice` method has been added in PS 1.7.7.2 and will cause an error for all lower versions as the method does not exists, as catched by @alejoasotelo.
See https://github.com/PrestaShop/PrestaShop/commit/e29e3077ba4610d0ac0ad260619c157e7369d41f

Reply:
Fixed

Reply:
The fix does not include stores with taxes excluded.
Maybe something like that?
```php
$summary = $this->context->cart->getSummaryDetails();
$isTaxExc = Group::getPriceDisplayMethod(Group::getCurrent()->id) == PS_TAX_EXC;
$totalPrice = $isTaxExc ? $summary['total_price_without_tax'] : $summary['total_price'];
$eventData = [
    'currency' => $this->context->currency->iso_code,
    'value' => (float) $totalPrice,
    'shipping_tier' => $carrierName,
    ];
];
```

Reply:
I am 99% sure that the value contains correct data = the same as tax excluded

Reply:
Just checked, taxes disabled, and the hook returns correct information. The rendered code doesn't get outputed for some reason, but I don't have the energy to investigate.